pkg/ccl/backupccl/backupdest/incrementals_test.go

```diff
@@ -11,12 +11,98 @@ package backupdest_test
 import (
 	"testing"
 
+	"github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backupdest"
 	"github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backuputils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/stretchr/testify/require"
 )
 
+func TestCollectionsAndSubdir(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	type testcase struct {
+		name                string
+		paths               []string
+		subdir              string
+		expectedCollections []string
+		expectedSubdir      string
+		expectedError       string
+	}
+	testcases := []testcase{
```

Comment:
nit: could you add a comment that says that new style show backup always expects a Subdir?